src/app/ods-config/components/data-source-form/data-source-form.component.ts

```diff
@@ -0,0 +1,71 @@
+/*
+Copyright (c) 2018 Friedrich-Alexander University Erlangen-Nürnberg (FAU)
+SPDX-License-Identifier: AGPL-3.0-only
+*/
+import {Component, OnInit} from '@angular/core';
+import {FormBuilder, FormGroup, Validators} from '@angular/forms';
+import {DataSource} from '../../../shared/model/data-source';
+import {DataSourceService} from '../../../shared/services/data-source.service';
+
+@Component({
+  selector: 'app-data-source-form',
+  templateUrl: './data-source-form.component.html',
+})
+export class DataSourceFormComponent implements OnInit {
+
+  dataSource: DataSource;
+  metaDataForm: FormGroup;
+  schemaForm: FormGroup;
+
+  constructor(private _formBuilder: FormBuilder,
+              private service: DataSourceService
+  ) {
+  }
+
+  ngOnInit() {
+    this.metaDataForm = this.createMetaDataForm();
+    this.schemaForm = this.createSchemaForm();
+  }
+
+  createMetaDataForm()  {
+    return this._formBuilder.group({
+      id: ['', Validators.required],
+      domainIdKey: [''],
+      metaData: this._formBuilder.group({
+        name: [''],
+        title: [''],
+        author: [''],
+        authorEmail: [''],
+        notes: [''],
+        url: [''],
+        termsOfUse: ['']
+      })
+    });
+  }
+
+  submitMetaData() {
+    if (this.metaDataForm.valid) {
+      this.dataSource = this.metaDataForm.getRawValue();
+    }
+    console.log(this.dataSource.id);
+    console.log(this.dataSource);
+  }
+
+  createSchemaForm() {
+    return this._formBuilder.group({
+      schema: ['']
+    });
+  }
+
+  submitDataSource() {
```

Comment:
What happens if input data is not valid?

Reply:
App should validate input before sending data. 
See commit : https://github.com/Keldami/ods-webclient/commit/c08b68c2096a7c9b8f03a047efb11e343edd0181